geolocation/ios/Plugin/GeolocationPlugin.swift

```diff
@@ -0,0 +1,200 @@
+        self.latitude = latitude
+        self.longitude = longitude
+    }
+}
+
+@objc(GeolocationPlugin)
+public class GeolocationPlugin: CAPPlugin, CLLocationManagerDelegate {
+
+    enum CallType {
+        case permissions
+        case singleUpdate
+        case watch
+    }
+
+    var locationManager = CLLocationManager()
+    private var callQueue: [String: CallType] = [:]
+
+    @objc func getCurrentPosition(_ call: CAPPluginCall) {
+        call.save()
+        callQueue[call.callbackId] = .singleUpdate
+
+        DispatchQueue.main.async {
+            self.locationManager.delegate = self
+            self.locationManager.requestWhenInUseAuthorization()
+
+            if call.getBool("enableHighAccuracy", false) == true {
+                self.locationManager.desiredAccuracy = kCLLocationAccuracyBest
+            } else {
+                self.locationManager.desiredAccuracy = kCLLocationAccuracyThreeKilometers
+            }
+
+            self.locationManager.requestLocation()
+        }
+    }
+
+    @objc func watchPosition(_ call: CAPPluginCall) {
+        call.save()
+        callQueue[call.callbackId] = .watch
+
+        DispatchQueue.main.async {
+            self.locationManager.delegate = self
+            self.locationManager.requestWhenInUseAuthorization()
+
+            if call.getBool("enableHighAccuracy", false) == true {
+                self.locationManager.desiredAccuracy = kCLLocationAccuracyBestForNavigation
+            } else {
+                self.locationManager.desiredAccuracy = kCLLocationAccuracyThreeKilometers
+            }
+
+            self.locationManager.startUpdatingLocation()
+        }
+    }
+
+    @objc func clearWatch(_ call: CAPPluginCall) {
+        guard let callbackId = call.getString("id") else {
+            CAPLog.print("Must supply id")
+            return
+        }
+        let savedCall = bridge?.getSavedCall(callbackId)
+        if savedCall != nil {
+            bridge?.releaseCall(savedCall!)
+
+            self.stopUpdating()
+        }
+        call.resolve()
+    }
+
+    public func locationManager(_ manager: CLLocationManager, didFailWithError error: Error) {
+        let removalQueue = callQueue.filter { $0.value == .permissions || $0.value == .singleUpdate }
+        callQueue = callQueue.filter { $0.value == .watch }
+
+        for (id, _) in removalQueue {
+            if let call = bridge?.getSavedCall(id) {
+                call.reject(error.localizedDescription)
+                bridge?.releaseCall(call)
+            }
+        }
+
+        for (id, _) in callQueue {
+            if let call = bridge?.getSavedCall(id) {
+                call.reject(error.localizedDescription)
+            }
+        }
+    }
+
+    public func locationManager(_ manager: CLLocationManager, didUpdateLocations locations: [CLLocation]) {
+        let removalQueue = callQueue.filter { $0.value == .singleUpdate }
+        callQueue = callQueue.filter { $0.value == .watch }
```

Comment:
It would be safer to test the inverse of the first predicate, otherwise calls could be lost if enum cases are ever added in the future. 
```suggestion
        callQueue = callQueue.filter { $0.value != .singleUpdate }
```